handlers/organization_test.go

```diff
@@ -99,3 +101,163 @@ func TestUnitCreateOrEditOrganization(t *testing.T) {
+		orgUUID := "org-uuid"
+
+		// Mock expected database interactions
+		mockDb.On("GetOrganizationByUuid", orgUUID).Return(db.Organization{OwnerPubKey: "test-key"}).Once()
+		mockDb.On("UpdateOrganizationForDeletion", orgUUID).Return(nil).Once()
+		mockDb.On("DeleteAllUsersFromOrganization", orgUUID).Return(nil).Once()
+		mockDb.On("ChangeOrganizationDeleteStatus", orgUUID, true).Return(db.Organization{Uuid: orgUUID, Deleted: true}).Once()
+
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(oHandler.DeleteOrganization)
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodDelete, "/organization/"+orgUUID, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		handler.ServeHTTP(rr, req)
+
+		assert.Equal(t, http.StatusOK, rr.Code)
+		mockDb.AssertExpectations(t)
+	})
+
+	t.Run("should handle failures in database updates", func(t *testing.T) {
+		orgUUID := "org-uuid"
+
+		// Mock database interactions with error
+		mockDb.On("GetOrganizationByUuid", orgUUID).Return(db.Organization{OwnerPubKey: "test-key"}).Once()
+		mockDb.On("UpdateOrganizationForDeletion", orgUUID).Return(errors.New("update error")).Once()
+
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(oHandler.DeleteOrganization)
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodDelete, "/organization/"+orgUUID, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		handler.ServeHTTP(rr, req)
+
+		assert.Equal(t, http.StatusInternalServerError, rr.Code)
+		mockDb.AssertExpectations(t)
+	})
+
+	t.Run("should set organization's deleted column to true", func(t *testing.T) {
+		orgUUID := "org-uuid"
+
+		// Mock the database interactions
+		mockDb.On("GetOrganizationByUuid", orgUUID).Return(db.Organization{OwnerPubKey: "test-key"}).Once()
+		mockDb.On("UpdateOrganizationForDeletion", orgUUID).Return(nil).Once()
+		mockDb.On("DeleteAllUsersFromOrganization", orgUUID).Return(nil).Once()
+		mockDb.On("ChangeOrganizationDeleteStatus", orgUUID, true).Return(db.Organization{Uuid: orgUUID, Deleted: true}).Once()
+
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(oHandler.DeleteOrganization)
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodDelete, "/organization/"+orgUUID, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		handler.ServeHTTP(rr, req)
+
+		// Asserting that the response status code is OK
+		assert.Equal(t, http.StatusOK, rr.Code)
+
+		// Decoding the response to check if Deleted field is true
+		var updatedOrg db.Organization
+		err = json.Unmarshal(rr.Body.Bytes(), &updatedOrg)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.True(t, updatedOrg.Deleted)
+
+		mockDb.AssertExpectations(t)
+	})
+
+	t.Run("should set Website, Github, and Description to null", func(t *testing.T) {
+		orgUUID := "org-uuid"
+
+		updatedOrg := db.Organization{
+			Uuid:        orgUUID,
+			OwnerPubKey: "test-key",
+			Website:     nil,
+			Github:      nil,
+			Description: nil,
```

Comment:
@AbdulWahab3181 change  all `nil` to empty strings `""`

Reply:
@elraphty But Our [ticket](https://github.com/stakwork/sphinx-tribes/issues/1334) scope is nill. Let me know if it still needs to be changed.

<img width="591" alt="image" src="https://github.com/stakwork/sphinx-tribes/assets/41957704/8a271c14-34f1-4bb9-a2ff-707e775f571a">

<img width="591" alt="image" src="https://github.com/stakwork/sphinx-tribes/assets/41957704/6ec451a3-99c3-47cb-94df-e5c37e13d997">

Reply:
Follow Raph's advice here @AbdulWahab3181

Reply:
Addressed

Reply:
> @elraphty But Our [ticket](https://github.com/stakwork/sphinx-tribes/issues/1334) scope is nill. Let me know if it still needs to be changed.
> 
> <img alt="image" width="591" src="https://private-user-images.githubusercontent.com/41957704/297841580-8a271c14-34f1-4bb9-a2ff-707e775f571a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NzI3NjgsIm5iZiI6MTcwNTY3MjQ2OCwicGF0aCI6Ii80MTk1NzcwNC8yOTc4NDE1ODAtOGEyNzFjMTQtMzRmMS00YmI5LWEyZmYtNzA3ZTc3NWY1NzFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDEzNTQyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNDIwNDc1NmJlZjY4YjU0NTZkM2U3ZWNkYzYwYWYwYmVjNWMwMDhkMGY3ZGYwYTEyMjcxMzQyOTliNDMzZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.u2GK8xWD0tQe-3spZWAIk4i-CN0gVGT4XcB4Kk784AA"> <img alt="image" width="591" src="https://private-user-images.githubusercontent.com/41957704/297841851-6ec451a3-99c3-47cb-94df-e5c37e13d997.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2NzI3NjgsIm5iZiI6MTcwNTY3MjQ2OCwicGF0aCI6Ii80MTk1NzcwNC8yOTc4NDE4NTEtNmVjNDUxYTMtOTljMy00N2NiLTk0ZGYtZTVjMzdlMTNkOTk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDEzNTQyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4OTFmNzk0Yzk5ZDIwNTMyMDNlNWNhMjAyZWYzMWIwNDZkYjAxYjM0MGM1ZDdjNmQyMjg2MmVjNzAxYTZiYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GHlkAINyvGHyjCnPs-Qx50olM73EKteEGywCYBnAZtA">

Yeah, but using nil may lead to GORM errors e.g

https://github.com/go-gorm/gorm/issues/6351
https://github.com/go-gorm/gorm/issues/3363